### PR TITLE
Data accessors for victory-bar

### PR DESCRIPTION
**Pull request by dandelany**
**WIP pull request, do not review/merge yet please!**

Adds data accessors to VictoryBar - see FormidableLabs/victory#84

This introduces another API change: previously, if the data was an array-of-arrays, it was assumed to be multiple data series and plotted as a grouped bar chart. However, with this change, an array-of-arrays is now a valid format for a _single_ data series (eg. `[[1, 2], [3, 4]]`). As a result, to plot a grouped bar chart from multiple data series, users must now explicitly pass a `grouped` prop, just like they are required to pass `stacked` for a stacked chart. If neither is passed, it will be treated as a single data series.

Also fixes issue with stacked bars #83.

### CODE REVIEW

**Comment by dandelany** [comment]
OK @boygirl this is ready for review!

File: src/components/victory-bar.jsx [hunk 1]
```diff
@@ -201,40 +194,75 @@ export default class VictoryBar extends React.Component {
...
+      CustomPropTypes.allOfType([CustomPropTypes.integer, CustomPropTypes.nonNegative]),
+      PropTypes.string,
+      PropTypes.arrayOf(PropTypes.string)
+    ])
   };
 
   static defaultProps = {
     data: defaultData,
     colorScale: "greyscale",
+    grouped: false,
```
Comment (boygirl) [inline src/components/victory-bar.jsx]: What do you think about leaving `grouped` undefined and writing a little `shouldGroup` method that returns `true` for arrays of arrays with the default accessors (_i.e._ `x: "x"`) and `false` else? I do think it's a little confusing to have your chart group even if you haven't set the `grouped` prop, but it turns this into a non-breaking change.

**Comment by boygirl** [comment]
cc/ @dandelany approved pending discussion on `grouped` prop

File: src/components/victory-bar.jsx (hunk 1 above)
Comment (dandelany) [inline src/components/victory-bar.jsx]: This seems pretty reasonable :+1: will add it in a few.

**Comment by dandelany** [comment]
OK @boygirl - added `shouldGroup` and a few more tests - please take another look. Thanks!

**Comment by boygirl** [comment]
:metal:

**Comment by dandelany** [comment]
Good catch thanks - removed. This should be ready to go now!

**Pull request merged**